firmware/src/gps.c

```diff
@@ -1,18 +1,74 @@
 #include "gps.h"
 
+// Static variables
+static uint32_t curr_time_ms;
+static uint32_t prev_time_ms;
+static uint8_t gps_connected_flag;
+static float utc_time;
+static GpsTime gps_time = { 0 };
+
+// Static function prototypes
+static uint32_t gpsTimeToMs(GpsTime time);
+
 ClockStatus Gps_Init(Gps *self)
 {
-    UNUSED(self);
+    self->timezone_offset = -5;
+
+    curr_time_ms = 0;
+    prev_time_ms = 0;
+    gps_connected_flag = 0;
+    utc_time = 0;
+
     return CLOCK_OK;
 }
-ClockStatus Gps_Connected(Gps *self)
+uint8_t Gps_Connected(Gps *self)
 {
-    UNUSED(self);
-    return CLOCK_OK;
+    // Check if connected flag is not set
+    gps_connected_flag = (uint8_t) self->gpsConnected();
+    if (!gps_connected_flag) {
+        return 0;
+    }
+
+    // Check if gps time has not changed
+    curr_time_ms = gpsTimeToMs(Gps_GetTime(self));
+    if (curr_time_ms == prev_time_ms) {
+        return 0;
+    }
+    prev_time_ms = curr_time_ms;
+
+    return 1;
 }
 GpsTime Gps_GetTime(Gps *self)
 {
-    UNUSED(self);
-    GpsTime zero = { 0 };
-    return zero;
+    // Get driver UTC time
+    utc_time = self->getUtcTime();
+
+    // Calculate gps hour, min and sec from UTC time and user inputted timezone
+    int8_t hr = ((uint8_t) utc_time / 10000) + self->timezone_offset;

```

Comment:
What is the time unit for utc_time? Based on your calculations it doesn't look like seconds or milliseconds

Reply:
Ah ok that makes more sense now lol